cargo-miri/bin.rs

```diff
@@ -595,8 +595,28 @@ fn phase_cargo_miri(mut args: env::Args) {
         }
     }
 
+    let metadata = get_cargo_metadata();
+
+    // Query cargo metadata for all the crates in this workspace, and if the caller hasn't already
+    // specified which crates we should consider local, add those to the local set.
+    // workspace members are emitted like "miri 0.1.0 (path+file:///path/to/miri)"
+    // Additionally, somewhere between cargo-metadata and TyCtxt, - gets replaced with _
+    assert!(metadata.workspace_members.len() > 0);
+    let mut flags = env::var("MIRIFLAGS").unwrap_or_default();
+    if !flags.contains("-Zmiri-local-crates=") {
+        flags += " -Zmiri-local-crates=";
+        for member in metadata.workspace_members {
+            let name = member.split(" ").nth(0).unwrap();
+            let name = name.replace("-", "_");
+            flags.push_str(&name);
+            flags.push(',');
+        }
+        flags.pop();
+    }
+    env::set_var("MIRIFLAGS", flags);
```

Comment:
This needs to pass data from `cargo-miri` into `miri`. It would be better engineering to have a private way to do this, but in the 1 hour or so that this code has existed I've already used the ability to set the "local" crates myself so maybe this is good actually?

Reply:
I think I would prefer a dedicated env var for this, rather than piggy-backing onto the (generally user-controlled) command-line arguments. We have quite a few private env vars already for communicating between our various binaries. (`cargo-miri` is at least 4 binaries packed into one.)